coil-base/src/main/java/coil/ImageLoaderBuilder.kt

```diff
@@ -179,26 +168,32 @@ class ImageLoaderBuilder(private val context: Context) {
             defaults = defaults,
             bitmapPoolSize = bitmapPoolSize,
             memoryCacheSize = memoryCacheSize,
-            okHttpClient = okHttpClient ?: buildDefaultOkHttpClient(),
+            callFactory = callFactory ?: buildDefaultOkHttpClient(),
             registry = registry ?: ComponentRegistry()
         )
     }
 
     private fun buildDefaultOkHttpClient(): OkHttpClient {
-        // Create the default image disk cache.
-        val cacheDirectory = Utils.getDefaultCacheDirectory(context)
-        val cacheSize = Utils.calculateDiskCacheSize(cacheDirectory)
-        val cache = Cache(cacheDirectory, cacheSize)
-
-        // Don't limit the number of requests by host.
-        val dispatcher = Dispatcher().apply {
-            maxRequestsPerHost = maxRequests
-        }
-
         return OkHttpClient.Builder()
-            .cache(cache)
-            .dispatcher(dispatcher)
-            .apply { okHttpClientBuilder?.invoke(this) }
+            .applyCoilOptimizations(context)
             .build()
     }
+
+    companion object {
+        /**
+         * Applies the preferred optimizations for Coil (image loading library) to this given Builder instance.
+         */
+        fun OkHttpClient.Builder.applyCoilOptimizations(context: Context): OkHttpClient.Builder = apply {
```

Comment:
Can you move this into `Utils`, set `Utils` to public, and set every other method to `internal`? I'd also change the name to `applyOkHttpClientOptimizations`.

Reply:
ad0d666